src/sidebar/components/thread-card.js

```diff
@@ -0,0 +1,65 @@
+import classnames from 'classnames';
+import { createElement } from 'preact';
+import { useCallback, useMemo } from 'preact/hooks';
+
+import debounce from 'lodash.debounce';
+import propTypes from 'prop-types';
+import useStore from '../store/use-store';
+import { withServices } from '../util/service-context';
+
+import Thread from './thread';
+
+/**
+ * A "top-level" `Thread`, rendered as a "card" in the sidebar. A `Thread`
+ * renders its own child `Thread`s within itself.
+ */
+function ThreadCard({ frameSync, settings, thread }) {
+  const threadTag = useMemo(() => {
```

Comment:
Perhaps a dumbish use of `useMemo`?

Reply:
Yes 😛-  There will be more overhead in `useMemo` here than is saved by eliminating a conditional property access.